chcemvediet/apps/inforequests/tests/test_cron.py

```diff
@@ -49,8 +50,8 @@ class UndecidedEmailReminderCronJobTest(CronTestCaseMixin, InforequestsTestCaseM
     Tests ``undecided_email_reminder()`` cron job.
     """
 
-    def _call_cron_job(self):
-        with mock.patch(u'chcemvediet.apps.inforequests.cron.workdays.between', side_effect=lambda a,b: (b-a).days):
+    def _call_cron_job(self, exception=None):
+        with mock.patch(u'chcemvediet.apps.inforequests.cron.workdays.between', side_effect=exception or (lambda a,b: (b-a).days)):
```

Comment:
Na simulovanie raisnutia exception sme pouzivali `nop()` funkciu, ktoru sme v teste v pripade potreby overridli mockom raisujucim exception. Vid ako je `nop()` pouzite v `poleno/mail/cron.py` a ako ho potom v `poleno/mail/tests/test_cron.py` overridujeme.

Takto s `nop()` je to IMHO prehladnejsie, nez upravovat funkcie, co robia aj nieco ine. A hlavne s `nop()` vieme jednoduchsie popisat zlozitejsie spravanie. Napriklad ak chceme, aby to raislo exception iba pri druhom zavolani z troch, mozeme pouzit `mock.patch(u'poleno.mail.cron.nop', side_effect=[None, Exception, None])`.

Reply:
@martinmacko47 V subore `cron.py` si funkciu `nop` vymazal, z toho som vchadzal ze ju tam nechces mat. Funkciu nop som vratil a testy nam zbehli, az na malu zmenu vo formatovani inforequestu.

Reply:
Kedze testy boli vypnute, tak som to neriesil. Popravte to `nop` sa mi moc nepaci. Ale nic ine rozume jednoduche ako v teste injectnut exception do kodu som nevymyslel. Mockovat ad-hoc pouzivane funkcie byva dost neprehladne. Takto s explicitnym `nop` je to o dost prehladnejsie.